examples/gno.land/r/gnoland/boards2/v1/permissions_default.gno

```diff
@@ -0,0 +1,273 @@
+package boards2
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/nt/commondao"
+
+	"gno.land/r/sys/users"
+)
+
+// DefaultPermissions manages users, roles and permissions.
+type DefaultPermissions struct {
+	superRole Role
+	dao       *commondao.CommonDAO
+	users     *avl.Tree // string(std.Address) -> []Role
+	roles     *avl.Tree // string(role) -> []Permission
+}
+
+// NewDefaultPermissions creates a new permissions type.
+// This type is a default implementation to handle users, roles and permissions.
+func NewDefaultPermissions(dao *commondao.CommonDAO) *DefaultPermissions {
+	if dao == nil {
+		panic("default permissions require a DAO")
+	}
+
+	dp := &DefaultPermissions{
+		dao:   dao,
+		roles: avl.NewTree(),
+		users: avl.NewTree(),
+	}
+
+	// Make sure that DAO users are indexed when DAO has members
+	dao.Members().IterateByOffset(0, dao.Members().Size(), func(m std.Address) bool {
+		dp.users.Set(m.String(), []Role(nil))
+		return false
+	})
+
+	return dp
+}
+
+// SetSuperRole assigns a super role.
+// A super role is one that have all permissions.
+// These type of role doesn't need to be mapped to any permission.
+func (dp *DefaultPermissions) SetSuperRole(r Role) {
+	dp.superRole = r
+}
+
+// AddRole add a role with one or more assigned permissions.
+func (dp *DefaultPermissions) AddRole(r Role, p Permission, extra ...Permission) {
+	dp.roles.Set(string(r), append([]Permission{p}, extra...))
+}
+
+// RoleExists checks if a role exists.
+func (dp DefaultPermissions) RoleExists(r Role) bool {
+	if dp.superRole != "" && r == dp.superRole {
+		return true
+	}
+
+	return dp.roles.Iterate("", "", func(name string, _ interface{}) bool {
+		return Role(name) == r
+	})
```

Comment:
```suggestion
	return dp.roles.Has(name)
```

Reply:
Ouch! Done in [b25b6d7](https://github.com/gnolang/gno/pull/2901/commits/b25b6d71083e313b3dff0f16903ef0e84b6c0786)